public/js/display.js

```diff
@@ -9,6 +9,30 @@ const MESSAGE_PERSIST_THRESHOLD = 5
 const REACTION_PREFIX =
  'reactions:message-channel-message-'
 
+// --- Helper functions for message expiration ---
+function calculateExpirationDate(message) {
+ // For negative scores, calculate when the message will expire
+ // A message expires when it reaches 0 score, which takes |score| hours at -1/hr velocity
+ const hoursToExpire = Math.abs(message.score)
+ const expirationDate = new Date()
+ expirationDate.setHours(expirationDate.getHours() + hoursToExpire)
+ return expirationDate
+}
+
+function formatExpirationDate(date) {
+ // Format as M/D/YYYY HH:MM AM/PM
+ const month = date.getMonth() + 1
+ const day = date.getDate()
+ const year = date.getFullYear()
+ let hours = date.getHours()
+ const minutes = date.getMinutes()
+ const ampm = hours >= 12 ? 'PM' : 'AM'
+ hours = hours % 12
+ hours = hours || 12 // Convert 0 to 12
```

Comment:
Do this as one step instead of two lines

```suggestion
 hours = hours % 12 || 12 // Convert 0 to 12
```